### DISCUSSION TITLE
Quest 2 ALVR Steam Linux?

### DISCUSSION THREAD

**Discussion by infidelus** [alvr-org/ALVR#1148] [Q&A]
Hi,

I've very recently purchased a Meta Quest 2, and am thoroughly enjoying playing Beat Saber & Eleven Table Tennis which came directly from the Meta store.  I would also love to be able to play some of the SteamVR titles in my library.

If I understand correctly, there is some support in ALVR for the Quest 2 on Linux (I'm using the recently released Linux Mint 21 if it matters), but I just can't seem to get it working.  Following the Linux section of the [installation guide](https://github.com/alvr-org/ALVR/wiki/Installation), I've installed the current version of SteamVR and downloaded the stable portable version of ALVR (v18.2.3) and have also tried a couple of recent nightly builds, but no matter what I do, when I launch ALVR all I end up with is:

![image](https://user-images.githubusercontent.com/39056969/183265326-5b319399-0cd1-4ddc-a597-6654d64cb8c0.png)

![image](https://user-images.githubusercontent.com/39056969/183265332-742c4ebf-5f46-41f3-95f5-0c16b5df8c36.png)

If others have it working, I'm guessing there's something I'm not doing right.

All help gratefully received.

**Comment by infidelus** [comment]
I guess nobody has anything to say to help.  I've managed to get SideQuest working and installed ALVR on to the Quest 2, but all it does is sit there saying 'waiting for server'.

**Reply by xDShot** [reply]
I just built both server and quest client

**Comment by infidelus** [comment]
I tried compiling using alvr_build_linux.sh all, but sadly all I get every time is:

> 2022-08-20 09:54:49 - Info: Preparing Linux Mint 21 (linuxmint) to build ALVR server
> ./alvr_build_linux.sh: line 174: prep_linuxmint_server: command not found
> 2022-08-20 09:54:49 - Critical Error: Failed to prepare Linux Mint 21 (linuxmint) for ALVR server build!
> 2022-08-20 09:54:49 - Info: Preparing Linux Mint 21 (linuxmint) to build ALVR client
> ./alvr_build_linux.sh: line 154: prep_linuxmint_client: command not found
> 2022-08-20 09:54:49 - Critical Error: Failed to prepare Linux Mint 21 (linuxmint) for ALVR client build!

**Reply by xDShot** [reply]
https://github.com/alvr-org/ALVR/wiki/Building-From-Source

**Reply by xDShot** [reply]
I don't build package

**Reply by ColdIce1605** [reply]
alvr_build_linux.sh only is works for a select group of distros it detects. so unless you want to edit the script to include your distro and you know how to do it, don't use that method

**Comment by infidelus** [comment]
Also didn't work as some of the packages are deprecated:

> Package gcc-8 is not available, but is referred to by another package.
> This may mean that the package is missing, has been obsoleted, or
> is only available from another source
> 
> E: Package 'gcc-8' has no installation candidate
> E: Package 'g++-8' has no installation candidate

I tried changing gcc and g++ to 9 instead of 8, which did then install but cargo doesn't work either:

> cargo xtask
> error: no such subcommand: `xtask`
> 
> 	Did you mean `test`?

**Reply by ColdIce1605** [reply]
make sure your running it that in the root of the project

**Reply by infidelus** [reply]
Thanks.  I could have sworn I was doing that, but clearly not.  It still fails though, albeit at a different point.  I'm guessing that may be due to the build dependencies that I can't install as they will remove other software I rely on.

**Reply by ColdIce1605** [reply]
Make sure to run 
cargo xtask prepare-deps --platform linux

**Reply by infidelus** [reply]
Thanks again.  I've managed to get the server to build now.

After trying  cargo xtask prepare-deps --platform linux, I started getting some other errors regarding missing dependencies, starting with a cuda error.  After installing nvidia-cuda-toolkit (which did remove some other nvidia packages but doesn't seem to have affected the desktop so far) I then got a yasm error so installed that too.  This, and a number of the other dependency errors came up, so I started going through the list of dependencies to see which ones I could install.  libjack-dev is the culprit that tries to remove wine-devel, so I didn't install that and couldn't install gcc-8 or g++-8 as they're not in the repository, but it looks like gcc-11 and g++-11 were installed at some point during this.

I had a very quick go at trying to build the client, but it errored out immediately (I do have android-tools-adb & android-tools-fastboot installed as I use them when I occasionally try other android roms), so I just grabbed the latest alvr-nightly and sideloaded that.

With the above done, I have managed to now connect and can get into SteamVR, though if I try and open the Steam store inside VR, I get an error:

![Screenshot from 2022-08-23 21-24-52](https://user-images.githubusercontent.com/39056969/186372629-d902335c-7f55-4659-b7e5-cce917c2ccd4.png)

I managed to get one random game that was in my Steam library to work (Paper Fire Rookie), but couldn't get Project Cars working (which the internet seems to suggest should work on the Quest 2 even though it's only got Rift support) as, when I launched it from Steam, all I got was a 3/4 size black screen in the ALVR landscape and then the whole screen started juddering horribly with every minor movement of my head.

[Edit]Running:

sudo apt install libxtst6:i386 libgtk2.0-0:i386 libpipewire-0.3-0:i386 libva-drm2:i386 libva-x11-2:i386 libvdpau1:i386

Seems to have fixed the Steam error (or at least I haven't seen it since), but none of the games I've tried to launch, with the exception of PFR, actually work.  Maybe I'm just unlucky with the ones I've tried so far.

**Reply by sol8712** [reply]
https://github.com/alvr-org/ALVR-nightly/releases/tag/v20.0.0-dev01%2Bnightly.2022.11.20
As ALVR progresses through development the Linux side sometimes breaks, the last version reported to work on linux is the nightly from Nov 20th.

Just set your bitrate to around 120 and your good to go. I haven't tried the audio or mic.

**Reply by infidelus** [reply]
Thanks.  I've pretty much given up on trying to get VR on the desktop at the moment.  While I have previously managed to connect to the headset I only ever got one game to work from my Steam library and that was one I just downloaded to test because it was free.  None of the other games that reportedly support VR work and all I ended up with is a nausea inducing juddering screen with the tiniest movement of my head.  

Since doing a fresh install of LM 21.1 I haven't even got as far as ALVR connecting to the headset.  None of the pre-built versions work and SteamVR always says there's no headset and continuously blocks the ALVR plugin.  I don't want to compile from source again on my main machine as the dependencies break packages I use and messes around with my NVIDIA drivers as can be seen from my previous efforts above to get it working.

I'm not totally giving up and will continue to follow the project in the hopes it someday works consistently on Linux, but it seems like the effort involved isn't worth it at this point in time.
